utils/netutils/timeout_conn.go

```diff
@@ -14,24 +14,42 @@ type timeoutConn struct {
 	net.Conn
 	sizeOfTransfer            int64
 	pool                      sync.Pool
+	workerPool                *pool
 	readTimeout, writeTimeout time.Duration
 }
 
 // newTimeoutConn returns a timeout conn wrapping around the provided one
-func newTimeoutConn(conn net.Conn, sizeOfTransfer int64, pool sync.Pool) *timeoutConn {
-	return &timeoutConn{Conn: conn, sizeOfTransfer: sizeOfTransfer, pool: pool}
+func newTimeoutConn(conn net.Conn, sizeOfTransfer int64, pool sync.Pool, workerPool *pool) *timeoutConn {
```

Comment:
Maybe `pool sync.Pool` should be called `byteBufferPool sync.Pool`? There are quite a lot of pools now and some way of differentiating them would be a good idea.